src/abstractdataframe/selection.jl

```diff
@@ -150,19 +150,29 @@ const TRANSFORMATION_COMMON_RULES =
     transformation and single column selection operations must be unique, so e.g.
     `select!(df, :a, :a => :a)` or `select!(df, :a, :a => ByRow(sin) => :a)` are not allowed.
 
-    As a general rule if `copycols=true` columns are copied and when
-    `copycols=false` columns are reused if possible. Note, however, that
-    including the same column several times in the data frame via renaming or
-    transformations that return the same object without copying may create
-    column aliases even if `copycols=true`. An example of such a situation is
-    `select!(df, :a, :a => :b, :a => identity => :c)`.
-    As a special case in `transform` and `transform!` column renaming always
-    copies columns to avoid storing aliased columns in the target data frame.
+    In general columns returned by transformations are stored in the target data
+    frame without copying. An exception to this rule is when columns from
+    the source data frame are reused in the target data frame (via expressions
+    like: `:x1`, `[:x1, :x2]`, `:x1 => :x2`, `:x1 => identity => :x2`, or
+    `:x1 => (x -> @view x[repeat(1, length(x))])`; note that in the last case
+    the source column is reused as its view is returned). In such cases the
+    behavior depends on the value of `copycols` keyword argument:
+    * if `copycols=true` then results of such transformations always perform a
+      copy of the source column or its view;
+    * if `copycols=false` then the result of the first transformation that reuses
+      the passed column directly (i.e. the result of the transformation returns
+      `true` when compared by `===` with the source column) gets stored in target
+      data frame without copying but every consecutive use of such a column performs
+      its copy (in other words the same column is not stored twice or more times
+      in the target data frame);
```

Comment:
```suggestion
    behavior depends on the value of the `copycols` keyword argument:
    * if `copycols=true` then results of such transformations always perform a
      copy of the source column or its view;
    * if `copycols=false` then copies are only performed to avoid storing the
      same column several times in the target data frame; more precisely, no
      copy is made the first time a column is used, but each subsequent
      reuse of a source column (when compared using `===`,
      which excludes views of source columns) performs a copy;
```